sdks/typescript/src/instrumentors/azure-ai-inference-otel-instrumentation.ts

```diff
@@ -0,0 +1,730 @@
+import {
+  InstrumentationBase,
+  InstrumentationNodeModuleDefinition,
+  InstrumentationConfig,
+  isWrapped,
+} from '@opentelemetry/instrumentation';
+import { trace, SpanStatusCode, SpanKind, Span as OtelSpan, context } from '@opentelemetry/api';
+import shimmer from 'shimmer';
+import { logger } from '../utils/logger';
+import { StreamWrapper, isAsyncIterable } from '../utils/stream-wrapper';
+import { isSDKShuttingDown } from '../shutdown';
+import type {
+  AzureInferenceChatCompletionParams,
+  AzureInferenceChatCompletionChunk,
+} from '../types/azure-ai-inference';
+
+// Import GenAI semantic conventions
+import {
+  SEMATTRS_GEN_AI_REQUEST_MODEL,
+  SEMATTRS_GEN_AI_SYSTEM,
+  SEMATTRS_GEN_AI_REQUEST_TEMPERATURE,
+  SEMATTRS_GEN_AI_REQUEST_MAX_TOKENS,
+  SEMATTRS_GEN_AI_REQUEST_TOP_P,
+  SEMATTRS_GEN_AI_RESPONSE_FINISH_REASONS,
+  SEMATTRS_GEN_AI_USAGE_INPUT_TOKENS,
+  SEMATTRS_GEN_AI_USAGE_OUTPUT_TOKENS,
+  SEMATTRS_GEN_AI_OPERATION_NAME,
+} from '../constants/gen-ai-semantic-conventions';
+
+const MODULE_NAME = '@azure-rest/ai-inference';
+const SUPPORTED_VERSIONS = ['*'];
+
+export interface AzureInferenceInstrumentationConfig extends InstrumentationConfig {
+  /**
+   * Hook called before request is sent
+   */
+  requestHook?: (span: OtelSpan, params: AzureInferenceChatCompletionParams) => void;
+
+  /**
+   * Hook called after response is received
+   */
+  responseHook?: (span: OtelSpan, response: any) => void;
+
+  /**
+   * Whether to use Proxy as fallback when shimmer fails
+   */
+  fallbackToProxy?: boolean;
+
+  /**
+   * Suppress internal instrumentation logs
+   */
+  suppressInternalInstrumentation?: boolean;
+}
+
+// Symbol to mark instrumented modules
+const LILYPAD_INSTRUMENTED = Symbol.for('lilypad.azure-ai-inference.instrumented');
+
+export class AzureInferenceInstrumentation extends InstrumentationBase {
+  constructor(config: AzureInferenceInstrumentationConfig = {}) {
+    super('@lilypad/instrumentation-azure-ai-inference', '0.1.0', config);
+  }
+
+  enable() {
+    const result = super.enable();
+
+    // Monkey-patch require to intercept module loading
+    const self = this;
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const Module = require('module');
+    const originalRequire = Module.prototype.require;
+
```

Comment:
Just noting this for attention, do we know that this will work for all the environments we want to support? I'm not an expert on different node require/resolution strategies but I know it can be tricky